ncm-download/src/main/perl/download.pm

```diff
@@ -248,11 +249,29 @@ sub get_remote_timestamp
     return $head->headers->last_modified();
 }
 
+sub _cleanup_tmpfile
```

Comment:
you could use `$self->cleanup` from `CAF::Path` instead of the method (i wasn't aware `CAF::Path` was already assumed).
if you want an error you can do
```perl
$self->error("Unable to delete temporary file $tmpfn: $self->{fail}") if (! defined($self->cleanup($tmpfn)));
```
instead this method